src/fabian_nemazi_ex/pa02/chutes_simulation.py

```diff
@@ -0,0 +1,179 @@
+
+
+class LazyPlayer(Player):
+    def __init__(self, board, dropped_steps=1):
+        super().__init__(board)
+        self.dropped_steps = dropped_steps
+
+        if dropped_steps >= 5:
+            raise ValuError(
+                "The input is invalid. "
+                "The player will never move forward "
+                "with a input higher than 5")
+
+    def move(self):
+        self.position += max(0, random.randint(1, 6) - self.dropped_steps)
+        self.position += self._board.position_adjustment(self.position)
+        self.moves += 1
+
+
+class Simulation:
+    def __init__(self, player_field, board=None, seed=12345,
+                 randomize_players=False):
+        self.player_field = player_field
+        self.player_types = frozenset(pc.__name__ for pc in player_field)
+        self.board = board if board is not None else Board()
+        self.randomize = randomize_players
+        self.results = []
+
+        random.seed(seed)
+
+    def single_game(self):
+        """
+        Returns winner type and number of steps for single game.
+
+        :returns: (number_of_steps, winner_class) tuple
+        """
+
+        players = [players(self.board) for players in self.player_field]
+        if self.randomize:
+            random.shuffle(players)
+
+        while True:
+            for player in players:
+                player.move()
+                if player._board.goal_reached(player.position):
+                    return player.moves, type(player).__name__
+
+    def run_simulation(self, number_of_games):
+        """
+        run_simulation() runs a given number of games and
+        stores the results in the Simulation object. It returns nothing."""
+        for _ in range(number_of_games):
+            self.results.append(self.single_game())
+
+    def get_results(self):
+        """
+        get_results() returns all results generated by
+        run_simulation() calls so far as a list of result tuples"""
+        return self.results
+
+    def winners_per_type(self):
+        """
+        winners_per_type() returns a dictionary
+        mapping player types to the number of wins,
+        """
+        winner_types = list(zip(*self.results))[1]
+        return {player_type: winner_types.count(player_type)
+                for player_type in self.player_types}
+
+    def durations_per_type(self):
+        """
+        durations_per_type() returns a dictionary mapping player types to
+        lists of game durations for that type"""
+        return {player_type: [win for win, types in self.results
+                              if types == player_type]
+                for player_type in self.player_types}
+
+    def players_per_type(self):
+        """
+        players_per_type returns a dictionary showing
+        how many players of each type participate
+        """
+        return {player_type.__name__: self.player_field.count(player_type)
+                for player_type in frozenset(self.player_field)}
```

Comment:
I think you made `frozenset(self.player_field)` an attribute in this class, why didn't you use it.